File: LuaLibs/src/Libs/FrameHandle/DaoJiShiTimer.lua
--[[
    @brief 倒计时定时器
]]
local M = GlobalNS.Class(GlobalNS.TimerItemBase)
GlobalNS["DaoJiShiTimer"] = M

function M:ctor()

end

function OnTimer(delta)
    if self.m_disposed then
        return;
    end

    self.m_curTime = self.m_curTime - delta;
    self.m_curLeftTimer = self.m_curLeftTimer + delta;

    if self.m_bInfineLoop then
        self.checkAndDisp();
    else
        if self.m_curTime <= 0 then
            self.disposeAndDisp();
        else
            self.checkAndDisp();
        end
    end
end

function reset()
    self.m_curTime = m_totalTime;
    self.m_curLeftTimer = 0;
    self.m_disposed = false;
end

return M

File: PyLibs/src/Libs/FrameHandle/DaoJiShiTimer.py
# -*- coding: utf-8 -*-

from Libs.FrameHandle.TimerItemBase import TimerItemBase;

'''
@brief 倒计时定时器
'''

class DaoJiShiTimer(TimerItemBase):
    
    def __init__(self):
        super(DaoJiShiTimer, self).__init__();
        
        self.mTypeId = "DaoJiShiTimer";

    
    def setTotalTime(self, value):
        super(DaoJiShiTimer, self).setTotalTime(value);
        self.mCurRunTime = value;


    def getRunTime(self):
        return self.mTotalTime - self.mCurRunTime;


    # 如果要获取剩余的倒计时时间，使用 getLeftCallTime
    def getLeftRunTime(self):
        return self.mCurRunTime;


    def OnTimer(self, delta):
        if (self.mIsDisposed):
            return;

        self.mCurRunTime -= delta;
        if(self.mCurRunTime < 0):
            self.mCurRunTime = 0;

        self.mIntervalLeftTime += delta;

        if (self.mIsInfineLoop):
            self.checkAndDisp();
        else:
            if (self.mCurRunTime <= 0):
                self.disposeAndDisp();
            else:
                self.checkAndDisp();



    def reset(self):
        self.mCurRunTime = self.mTotalTime;
        self.mCurCallTime = 0;
        self.mIntervalLeftTime = 0;
        self.mIsDisposed = False;

